packages/evo-marko/src/tags/evo-eek/index.marko

```diff
@@ -0,0 +1,89 @@
+import type { AttrString } from "marko/tags-html";
+export interface Input extends Marko.HTML.Div {
+  max: string;
+  min: string;
+  rating: string;
+  /**
+   *  The aria-label accessibility label for the eek rating. This is for internationalization.
+   *  It should use eekRating, min, and max values in the label in order to demonstrate
+   *  to screen readers the content on the component.
+   *  Expected return `Energy Rating: ${eekRating}. Range: ${max} - ${min}.`
+   **/
+  "aria-label"?: AttrString;
+  size?: "regular" | "large";
+}
+static const validRanges = {
+  "A+++": ["D", "E", "G"],
+  "A++": ["E", "G"],
+  "A+": ["F", "G"],
+  A: ["G"],
+};
+static function eekUtil(input: Input) {
+  const { max, min, rating } = input;
+  const validMax = validRanges[max as keyof typeof validRanges];
+  if (!(validMax && validMax.indexOf(min) > -1)) {
+    return null;
+  }
+  // Count from max until you reach rating
+  let current = max;
+  let i = 1;
+  while (current !== rating) {
+    i++;
+    if (min === current) {
+      // It's at end, didn't match rating
+      return null;
+    } else if (current.length > 1) {
+      // Has + in string, remove one plus
+      current = current.slice(0, current.length - 1);
+    } else {
+      // No pluses get next character code
+      current = String.fromCharCode(current.charCodeAt(0) + 1);
+    }
+  }
+  return i > 7 ? 7 : i;
+}
+<const/{
+  class: inputClass,
+  size,
+  max,
+  min,
+  rating,
+  // ...htmlInput
+}=input/>
+
+<const/htmlInput=(({
+  class: inputClass,
+  size,
+  max,
+  min,
+  rating,
+  ...rest
+}) => rest)(input)/>
+
+<const/eekRating=eekUtil(input)/>
```

Comment:
Not a huge deal, but passing in all of `input` like this is a deopt so Marko will serialize the whole thing instead of just the important parts. I think `eekUtil` should take `min`, `max`, and `rating` as params which are passed individually